src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+    this.chipVersion = chipVersion;
+  };
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+}
 
-class DeliveryDrone {}
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x = 0, y = 0, z = 0 }, chipVersion) {
+    super(name, weight, {
+      x, y, z,
+    },
+    chipVersion);
+    this.chipVersion = chipVersion;
+
+    this.coords = {
+      x,
+      y,
+      z,
+    };
+  }
```

Comment:
Не обязательно в конструктор передавать все координаты.
можно сделать так:
```suggestion
  constructor(name, weight, coords, chipVersion) {
    super(name, weight, coords, chipVersion);

    this.coords.z = coords.z || 0;
  }
```